autoscheduler/frontend/src/utils/throttle.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Function to throttles the execution of other functions
+ * @param id ID to use, each ID operates on a separate timer
+ * @param callback Callback to run after delay expires
+ * @param delay Delay (in ms) before running effect
+ * @param runOtherIDs Whether to immediately run pending callbacks for IDs other than specified id
+ */
+const throttle = ((): (id: string, callback: any, delay: number, runOtherIDs: boolean) => void => {
+  const intervals = new Map<string, number>();
+  const intervalTimes = new Map<string, number>();
+  const functions = new Map<string, any>();
+
+  return (id: string, callback: any, delay: number, runOtherIDs = false): void => {
+    // Update interval frequency if needed
+    if (delay !== intervalTimes.get(id)) {
```

Comment:
When would the delay stored in `intervalTimes` not match the `delay` passed as a parameter?

Reply:
This happens in a couple situations:
- The first time it's called, the get returns `undefined` and this block is run to set the initial interval
- If you change the delay in the code that calls `throttle`, this will update the interval to the correct timer (this isn't being done right now, but doing it this way gives more flexibility)